common/src/main/java/tachyon/util/ShellUtils.java

```diff
@@ -0,0 +1,246 @@
+   */
+  public static class ExitCodeException extends IOException {
+
+    private static final long serialVersionUID = -6520494427049734809L;
+
+    private final int mExitCode;
+
+    public ExitCodeException(int exitCode, String message) {
+      super(message);
+      mExitCode = exitCode;
+    }
+
+    public int getExitCode() {
+      return mExitCode;
+    }
+
+    @Override
+    public String toString() {
+      final StringBuilder sb = new StringBuilder("ExitCodeException ");
+      sb.append("exitCode=").append(mExitCode).append(": ");
+      sb.append(super.getMessage());
+      return sb.toString();
+    }
+  }
+
+  public interface CommandExecutor {
+
+    void execute() throws IOException;
+
+    int getExitCode() throws IOException;
+
+    String getOutput() throws IOException;
+
+    void close();
+
+  }
+
+  /**
+   * A simple shell command executor.
+   *
+   * <code>ShellCommandExecutor</code>should be used in cases where the output of the command needs
+   * no explicit parsing and where the command, working directory and the environment remains
+   * unchanged. The output of the command is stored as-is and is expected to be small.
+   */
+  public static class ShellCommandExecutor extends ShellUtils implements CommandExecutor {
+
+    private String[] mCommand;
+    private StringBuffer mOutput;
+
+    public ShellCommandExecutor(String[] execString) {
+      mCommand = execString.clone();
+    }
+
+    /** Execute the shell command. */
+    public void execute() throws IOException {
+      run();
+    }
+
+    @Override
+    public String[] getExecString() {
+      return mCommand;
+    }
+
+    @Override
+    protected void parseExecResult(BufferedReader lines) throws IOException {
+      mOutput = new StringBuffer();
+      char[] buf = new char[512];
+      int nRead;
+      while ((nRead = lines.read(buf, 0, buf.length)) > 0) {
+        mOutput.append(buf, 0, nRead);
+      }
+    }
+
+    /** Get the output of the shell command. */
+    public String getOutput() {
+      return (mOutput == null) ? "" : mOutput.toString();
+    }
+
+    /**
+     * Returns the commands of this instance. Arguments with spaces in are presented with quotes
+     * round; other arguments are presented raw.
+     * @return a string representation of the object
+     */
+    @Override
+    public String toString() {
+      StringBuilder builder = new StringBuilder();
+      String[] args = getExecString();
+      for (String s : args) {
+        if (s.indexOf(' ') >= 0) {
+          builder.append('"').append(s).append('"');
+        } else {
+          builder.append(s);
+        }
+        builder.append(' ');
+      }
+      return builder.toString();
+    }
+
+    @Override
+    public void close() {}
+  }
+
+  /**
+   * Static method to execute a shell command. Covers most of the simple cases without requiring the
+   * user to implement the <code>Shell</code> interface.
+   * @param cmd shell command to execute
```

Comment:
insert newline between method-level javadoc and parameter / return argument descriptions